### PR TITLE
[Redesign] Accessibility fixes, SVG, and other polish

### PR DESCRIPTION
**Pull request by joelverhagen**
1. Use `<section>` instead of `<div>` for main container
1. Use sm grid instead of md grid
1. Add roles to navbar
1. Use SVG and fall back to PNG
1. Darken the warning orange. This improves the ratio but does not resolve the Keros warning. I think this is okay since it's not an English character or word and just a roughly identifiable shape.
1. Fix search button pixel issue on Firefox 
1. Fix padding issue in the DEV/INT banner causing horizontal scroll

/cc @microsoftsam

### CODE REVIEW

File: src/NuGetGallery/Views/Shared/Gallery/Header.cshtml [hunk 1]
```diff
@@ -2,53 +2,73 @@
...
             </div>
         </div>
     </div>
 }
 
-<nav class="navbar navbar-inverse">
+@helper DisplayNavigationItem(string tab, string url, bool bold = false, string title = null)
+{
+    <li class="@(ViewBag.Tab == tab ? "active" : string.Empty)"
+        aria-selected="@(ViewBag.Tab == tab ? "true" : "false")">
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Shared/Gallery/Header.cshtml]: add role="none" on the li

File: src/NuGetGallery/Views/Users/ForgotPassword.cshtml [hunk 2]
```diff
@@ -25,7 +25,7 @@
     }
 }
 
-<div class="container main-container page-forgot-password">
+<section class="container main-container page-forgot-password">
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Users/ForgotPassword.cshtml]: Add role="main"

File: src/NuGetGallery/Views/Users/PasswordSent.cshtml [hunk 3]
```diff
@@ -3,7 +3,7 @@
     Layout = "~/Views/Shared/Gallery/Layout.cshtml";
 }
 
-<div class="container main-container">
+<section class="container main-container">
     <h1>Password Reset Sent</h1>
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Users/PasswordSent.cshtml]: Add role="main"

File: src/NuGetGallery/Views/Users/PasswordSent.cshtml [hunk 4]
```diff
@@ -3,7 +3,7 @@
     Layout = "~/Views/Shared/Gallery/Layout.cshtml";
 }
 
-<div class="container main-container">
+<section class="container main-container">
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Users/PasswordSent.cshtml]: Add role="main"

File: src/NuGetGallery/Views/Shared/Gallery/Header.cshtml [hunk 5]
```diff
@@ -2,53 +2,73 @@
...
-                    <a href="@Url.Home()"><img class="navbar-logo" src="~/Content/gallery/img/logo-94x27.png" alt="Smaller NuGet logo in header" /></a>
+                    <a href="@Url.Home()"><img class="navbar-logo" width="94" height="27" alt="Smaller NuGet logo in header"
+                                               src="~/Content/gallery/img/logo.svg"
+                                               @ViewHelpers.ImageFallback(Url.Absolute("~/Content/gallery/img/logo-94x27.png")) /></a>
                 </div>
                 <div id="navbar" class="navbar-collapse collapse">
-                    <ul class="nav navbar-nav">
-                        <li class="@(ViewBag.Tab == "Packages" ? "active" : string.Empty)"><a href="@Url.PackageList()"><span>Packages</span></a></li>
-                        <li class="@(ViewBag.Tab == "Upload" ? "active" : string.Empty)"><a href="@Url.UploadPackage()"><span>Upload</span></a></li>
+                    <ul class="nav navbar-nav" role="menu">
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Shared/Gallery/Header.cshtml]: Add role="menubar" to the div id="navbar"

File: src/NuGetGallery/Views/Shared/Gallery/Header.cshtml (hunk 1 above)
Comment (microsoftsam) [inline src/NuGetGallery/Views/Shared/Gallery/Header.cshtml]: or role="presentation"

File: src/NuGetGallery/Views/Shared/Gallery/Header.cshtml (hunk 5 above)
Comment (joelverhagen) [inline src/NuGetGallery/Views/Shared/Gallery/Header.cshtml]: Ah thanks, missed that one.

File: src/NuGetGallery/Views/Users/PasswordSent.cshtml (hunk 4 above)
Comment (joelverhagen) [inline src/NuGetGallery/Views/Users/PasswordSent.cshtml]: Fixed.

File: src/NuGetGallery/Views/Shared/Gallery/Header.cshtml (hunk 5 above)
Comment (joelverhagen) [inline src/NuGetGallery/Views/Shared/Gallery/Header.cshtml]: Fixed.

File: src/NuGetGallery/Views/Users/ForgotPassword.cshtml (hunk 2 above)
Comment (joelverhagen) [inline src/NuGetGallery/Views/Users/ForgotPassword.cshtml]: Fixed.

File: src/NuGetGallery/Views/Shared/Gallery/Header.cshtml (hunk 1 above)
Comment (joelverhagen) [inline src/NuGetGallery/Views/Shared/Gallery/Header.cshtml]: Fixed.

File: src/NuGetGallery/Views/Authentication/SignIn.cshtml [hunk 6]
```diff
@@ -5,45 +5,54 @@
...
+            <div class="col-sm-12">
+                <p>
+                    @if (!string.IsNullOrEmpty(provider.UI.SignInInfo))
+                    {
+                        @provider.UI.SignInInfo
+                    }
+                    else
+                    {
+                        <text>
+                            We recommend using a @(provider.UI.AccountNoun ?? "external account") to sign in to NuGet.org.
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Authentication/SignIn.cshtml]: If `provider.UI.AccountNoun` is null, the message should be "an external account" not "a external account".

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml [hunk 7]
```diff
@@ -29,15 +29,16 @@
...
 
-<div class="container main-container page-package-details">
+<section role="main" class="container main-container page-package-details">
     <div class="row">
-        <div class="col-md-1">
+        <aside class="col-sm-1">
             <h3>
-                <img class="img-responsive" alt="Icon for package @Model.Id" onerror="this.src = '@Url.Absolute("~/Content/gallery/img/default-package-icon.png")';"
-                     src="@(PackageHelper.ShouldRenderUrl(Model.IconUrl) ? Model.IconUrl : Url.Absolute("~/Content/gallery/img/default-package-icon.png"))" />
+                <img class="img-responsive" alt="Icon for package @Model.Id"
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackage.cshtml]: I think this should have an `aria-hidden="true"` attribute as it is a decorative image. @microsoftsam can you confirm this?

File: src/NuGetGallery/Views/Shared/Gallery/Footer.cshtml [hunk 8]
```diff
@@ -1,35 +1,39 @@
...
                 <p>
                     Read the Frequently Asked Questions about NuGet and see if your question made the list.
                 </p>
             </div>
         </div>
         <div class="row row-gap">
-            <div class="col-md-3">
-                <img src="~/Content/gallery/img/logo-184x53.png" alt="Larger NuGet logo in the page footer" />
+            <div class="col-sm-3">
+                <img alt="Larger NuGet logo in the page footer" width="184" height="53"
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Shared/Gallery/Footer.cshtml]: This image is also decorative and may also need a `aria-hidden="true"` attribute.

File: src/NuGetGallery/Views/Authentication/SignIn.cshtml (hunk 6 above)
Comment (joelverhagen) [inline src/NuGetGallery/Views/Authentication/SignIn.cshtml]: Heh, good point.

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml (hunk 7 above)
Comment (joelverhagen) [inline src/NuGetGallery/Views/Packages/DisplayPackage.cshtml]: I will align with what you have on the package search page.

Comment (joelverhagen) [inline src/NuGetGallery/Views/Packages/DisplayPackage.cshtml]: Done.

File: src/NuGetGallery/Views/Shared/Gallery/Footer.cshtml (hunk 8 above)
Comment (joelverhagen) [inline src/NuGetGallery/Views/Shared/Gallery/Footer.cshtml]: Done.

**Comment by microsoftsam** [comment]
Doesn't need to be part of this change, but the search button needs an aria-label: ```aria-label="Search"``` so Narrator reads out "search button"

**Comment by joelverhagen** [comment]
> Doesn't need to be part of this change, but the search button needs an aria-label: ```aria-label="Search"``` so Narrator reads out "search button"

This is already a "miscellaneous" PR so I'll do it here 😄

**Comment by joelverhagen** [comment]
🔔 👀 🙏

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml [hunk 9]
```diff
@@ -29,15 +29,17 @@
...
+<section role="main" class="container main-container page-package-details">
     <div class="row">
-        <div class="col-md-1">
+        <aside class="col-sm-1">
             <h3>
-                <img class="img-responsive" alt="Icon for package @Model.Id" onerror="this.src = '@Url.Absolute("~/Content/gallery/img/default-package-icon.png")';"
-                     src="@(PackageHelper.ShouldRenderUrl(Model.IconUrl) ? Model.IconUrl : Url.Absolute("~/Content/gallery/img/default-package-icon.png"))" />
+                <img class="img-responsive"
+                     aria-hidden="true" alt=""
+                     onerror="this.src = '@Url.Absolute("~/Content/gallery/img/default-package-icon.svg")'; this.onerror = null;"
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackage.cshtml]: Use the `ImageFallback` helper here.

File: src/NuGetGallery/Views/Pages/Home.cshtml [hunk 10]
```diff
@@ -42,23 +46,23 @@
             </p>
         </div>
         <div class="row row-gap">
-            <div class="col-md-4">
+            <div class="col-sm-4">
                 <img src="~/Content/gallery/img/upload-hero.png" alt="Cartoon image of a person lifting a package into the clouds" />
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Pages/Home.cshtml]: Do we really want these `alt` on these hero images? These are purely decorative, so maybe `aria-hidden` would be better?

Comment (joelverhagen) [inline src/NuGetGallery/Views/Pages/Home.cshtml]: These sort of act as heading for the 3 columns so I don't think they are purely decorative.

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml (hunk 9 above)
Comment (joelverhagen) [inline src/NuGetGallery/Views/Packages/DisplayPackage.cshtml]: Nice catch.

**Pull request merged**